vignettes/classes-objects.Rmd

```diff
@@ -249,7 +270,103 @@ x@length <- 5
 x
 ```
 
-A `setter` is a function with arguments `self` and `value` that returns a modified object.
+### Common Patterns
+
+`getter`, `setter`, `default`, and `validator` can be used to implement many common patterns of properties.
+
+#### Deprecated properties 
+
+A `setter` + `getter` can be used to to deprecate a property:
+
+```{r}
+Person <- new_class("Person", properties = list(
+ first_name = class_character,
+ firstName = new_property(
```

Comment:
Shouldn't this `firstName` property be of `class_character`? Even though it is deprecated, we still want it to function properly. And I guess we would really want it to be a scalar. What happens if there is no reasonable default, as in this case? Could the default be made to be `quote(first_name)`? Then the setter avoids a warning if the value is the same as `first_name`.